doc/ansible/dev/apb_migration_guide.md

```diff
@@ -0,0 +1,241 @@
+- `_apb_plan_id`: Operators have no concept of a plan
+- `_apb_service_class_id`: This concept is replaced by the group/version/kind specified in your CRD
+- `_apb_service_instance_id`: This concept is replaced by `meta.name`, the name of the Custom Resource created by the user requesting the action.
+- `_apb_last_requesting_user`: There is no analogue to this.
+- `_apb_provision_creds`: There is no analogue to this.
+- `_apb_service_binding_id`: This concept is replaced by the `meta.name` of a `<kind>Binding` resource
+- `namespace`: This is accessible via the `meta.namespace` variable
+
+Instead, the Ansible Operator will pass in a field called `meta`, which contains the `name` and `namespace` of the Custom Resource that the user created.
+
+
+#### asb_encode_binding
+This module will not be present in the Ansible Operator base image. In order to save credentials after a successful provision, you will need to create a `secret` in Kubernetes, and update the status of your custom resource so that people can find it. For example, if we have the following Custom Resource group/version/kind:
+
+```yaml
+version: v1alpha1
+group: apps.example.com
+kind: PostgreSQL
+```
+
+the following task:
+
+```yaml
+- name: encode bind credentials
+  asb_encode_binding:
+    fields:
+      DB_TYPE: postgres
+      DB_HOST: "{{ app_name }}"
+      DB_PORT: "5432"
+      DB_USER: "{{ postgresql_user }}"
+      DB_PASSWORD: "{{ postgresql_password }}"
+      DB_NAME: "{{ postgresql_database }}"
+```
+
+would become:
+
+```yaml
+- name: Create bind credential secret
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: Secret
+      metadata:
+        name: '{{ meta.name }}-credentials'
+        namespace: '{{ meta.namespace }}'
+      data:
+        DB_TYPE: "{{ 'postgres' | b64encode }}"
+        DB_HOST: "{{ app_name | b64encode }}"
+        DB_PORT: "{{ '5432' | b64encode }}"
+        DB_USER: "{{ postgresql_user | b64encode }}"
+        DB_PASSWORD: "{{ postgresql_password | b64encode }}"
+        DB_NAME: "{{ postgresql_database | b64encode }}"
+
+- name: Attach secret to CR status
+  k8s_status:
+    api_version: apps.example.com/v1alpha1
+    kind: PostgreSQL
+    name: '{{ meta.name }}'
+    namespace: '{{ meta.namespace }}'
+    status:
+      bind_credentials_secret: '{{ meta.name }}-credentials'
+```
+
+#### ansible_kubernetes_modules
+* The ansible_kubernetes_modules role and the generated modules are now deprecated.
+* The `k8s` module was added in Ansible 2.6 and is the supported way to interact with Kubernetes from Ansible.
+* The `k8s` module takes normal kubernetes manifests, so if you currently rely on the old generated modules some refactoring will be required.
+
+
+# Terms
+kind
+group
+apiVersion
+watches.yaml
+finalizer
+CRD
+
+# convert.py
+This script should be run from inside the APB directory, next to the `apb.yml`
```

Comment:
s/This script/The `convert.py` script/

Reply:
I would like to see an example of this being run, what should I expect when I run this?